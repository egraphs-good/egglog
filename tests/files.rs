use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
