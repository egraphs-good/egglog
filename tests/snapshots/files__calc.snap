---
source: tests/files.rs
expression: serialized
---
nodes:
  I-0:
    op: I
    children: []
    eclass: "0"
    cost: 9223372036854776000
  A4-0:
    op: A4
    children: []
    eclass: "4"
    cost: 9223372036854776000
  b-0:
    op: b
    children: []
    eclass: "7"
    cost: 9223372036854776000
  A2-0:
    op: A2
    children: []
    eclass: "3"
    cost: 9223372036854776000
  A-0:
    op: A
    children: []
    eclass: "1"
    cost: 9223372036854776000
  B-0:
    op: B
    children: []
    eclass: "2"
    cost: 9223372036854776000
  A8-0:
    op: A8
    children: []
    eclass: "5"
    cost: 9223372036854776000
  a-0:
    op: a
    children: []
    eclass: "6"
    cost: 9223372036854776000
  g*-6828067974578293639:
    op: g*
    children:
      - A-0
      - A-0
    eclass: "3"
    cost: 1
  g*-15443823971188422721:
    op: g*
    children:
      - g*-6828067974578293639
      - A2-0
    eclass: "4"
    cost: 1
  g*-5435976351651060604:
    op: g*
    children:
      - g*-15443823971188422721
      - A4-0
    eclass: "5"
    cost: 1
root_eclasses: []
class_data:
  "0":
    type: G
  "4":
    type: G
  "7":
    type: G
  "3":
    type: G
  "1":
    type: G
  "2":
    type: G
  "5":
    type: G
  "6":
    type: G

